lamindb/models/history.py

```diff
@@ -0,0 +1,89 @@
+from django.db import models
+
+
+class HistoryTableState(models.Model):
+    """A list of tables for which we're tracking history.
+
+    This table serves two purposes: it allows us to store the
+    name of the table for a given history record with less state.
+
+    NOTE: This table only makes sense for a single LaminDB instance,
+    since the ID of a given table may be different on different instances.
+    You should join this table into History to determine which table
+    a given history record applies to, rather than using the IDs directly.
+    """
+
+    # Using a smallserial here because it's incredibly unlikely that
+    # we'll have more than 32k tables
+    id = models.SmallAutoField(primary_key=True)
+    table_name = models.CharField(max_length=255)
+    backfilled = models.BooleanField()
+
+
+class HistoryMigrationState(models.Model):
+    """A summary of the state of Django's migrations when a history record was recorded.
+
+    When a history record is recorded, we need to record the state of the data migrations
+    for each Django app so that we know if we need to migrate the record from the version
+    it had when it was written to the version that exists in the target during replay.
+    """
+
+    # Using a smallserial here because it's incredibly unlikely that
+    # we'll have more than 32k schema migrations.
+    id = models.SmallAutoField(primary_key=True)
+    migration_history_id = models.JSONField()
+
+
+class History(models.Model):
+    """Stores the edit history for all LaminDB tables."""
+
+    seqno = models.AutoField(primary_key=True)
+    history_migration_state = models.ForeignKey(
+        HistoryMigrationState, on_delete=models.PROTECT
+    )
+    table = models.ForeignKey(HistoryTableState, on_delete=models.PROTECT)
+    id = models.UUIDField()
```

Comment:
Can we call this `uid` and make it a `base62` string? I added code for this up there: https://github.com/laminlabs/lamindb/pull/2642/files#r2081960745

The user facing links to things are always based on a `uid` similar to how stripe or modal or other platforms do it.

This really is mostly a design/aesthetic decision (sharing an id like `3ixi4FetqaJk` and links like this https://lamin.ai/vitessce/examples/transform/3ixi4FetqaJk is just a bit more user friendly)

Sometimes it's also nice that one can encode information in some of the chars more easily than in the UUID.

And 16 byte vs. 18 byte don't make a difference.